src/parser/template.rs

```diff
@@ -30,23 +32,160 @@ impl<'a> From<Pair<'a, Rule>> for Template {
                 _ => unreachable!(),
             }
         }
         TemplateLine {
             indentation,
-            content,
+            segments,
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::parser::{parse, segment::Segment};
+
+    #[test]
+    fn empty_template() {
+        let templates = parse("main =\n----").unwrap();
+
+        assert_eq!(
+            templates,
+            &[Template {
+                name: "main".into(),
+                indent_ignored: 4,
+                lines: vec![]
+            }]
+        );
+    }
+
+    #[test]
+    fn space_indented_content() {
+        let templates = parse("main =\n    indent4\n     indent5\n----").unwrap();
+
+        assert_eq!(
+            templates,
+            &[Template {
+                name: "main".into(),
+                indent_ignored: 4,
+                lines: vec![
+                    TemplateLine {
+                        indentation: "    ".into(),
+                        segments: vec![Segment::Content("indent4".into())],
+                    },
+                    TemplateLine {
+                        indentation: "     ".into(),
+                        segments: vec![Segment::Content("indent5".into())],
+                    }
+                ]
+            }]
+        );
+    }
+
+    #[test]
+    fn tab_indented() {
+        let templates = parse("main =\n\tindent1\n\t\tindent2\n-").unwrap();
+
+        assert_eq!(
+            templates,
+            &[Template {
+                name: "main".into(),
+                indent_ignored: 1,
+                lines: vec![
+                    TemplateLine {
+                        indentation: "\t".into(),
+                        segments: vec![Segment::Content("indent1".into())],
+                    },
+                    TemplateLine {
+                        indentation: "\t\t".into(),
+                        segments: vec![Segment::Content("indent2".into())],
+                    }
+                ]
+            }]
+        );
+    }
+
+    #[test]
+    fn mixed_indentation() {
+        let templates = parse("main =\n    \tindent\n----").unwrap();
+
+        assert_eq!(
+            templates,
+            &[Template {
+                name: "main".into(),
+                indent_ignored: 4,
+                lines: vec![TemplateLine {
+                    indentation: "    \t".into(),
+                    segments: vec![Segment::Content("indent".into())],
+                }]
+            }]
+        );
+    }
+
+    #[test]
+    fn ignore_more_indentation() {
+        let templates = parse("main =\n  content\n----").unwrap();
+
+        assert_eq!(
+            templates,
+            &[Template {
+                name: "main".into(),
+                indent_ignored: 4,
+                lines: vec![TemplateLine {
+                    indentation: "  ".into(),
+                    segments: vec![Segment::Content("content".into())],
```

Comment:
@jordwest if decide that `----` will ignore 4 characters, rather than just whitespace, then this line should be changed to 

```rs
segments: vec![Segment::Content("ntent".into())],
```